containers/test-apps/kitchen/bin/kitchen

```diff
@@ -741,6 +755,25 @@ class Kitchen(HTTPWebSocketsHandler):
             if str(name).startswith('x-kitchen-trailer-'):
                 self.__response_trailers[name[len('x-kitchen-trailer-'):]] = value
 
+        # Process default response status
+        default_status_value = self.headers.get('x-kitchen-default-status-value')
+        if default_status_value is not None:
+            if _allow_response_status_change:
+                self.logger().info('Changing default response status to {}'.format(default_status_value))
+                _default_response_status = int(default_status_value)
+                default_timeout = self.headers.get('x-kitchen-default-status-timeout')
+                if default_timeout is not None:
+                    default_timeout_ms = int(default_timeout)
+                    if default_timeout_ms > 0:
+                        self.logger().info('Will reset default response status after {} ms'.format(default_timeout_ms))
+                        run_after_ms(default_timeout_ms, reset_default_response_status, self)
+            else:
+                self.logger().info('Ignoring request to change the default response status')
+
+        # Set the response status if it is not set
+        if self.__status is None:
+            self.__status = _default_response_status
```

Comment:
I really liked the idea of moving this into `send_response`, but this works too.